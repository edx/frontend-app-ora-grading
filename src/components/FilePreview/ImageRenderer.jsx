import React from 'react';
import PropTypes from 'prop-types';

const ImageRenderer = ({ url, fileName }) => (<img alt={fileName} className="image-renderer" src={url} />);

ImageRenderer.defaultProps = {
  fileName: '',
};

ImageRenderer.propTypes = {
  url: PropTypes.string.isRequired,
  fileName: PropTypes.string,
};

export default ImageRenderer;
