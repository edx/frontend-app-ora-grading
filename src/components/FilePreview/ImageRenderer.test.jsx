import React from 'react';
import { shallow } from 'enzyme';

import ImageRenderer from './ImageRenderer';

describe('Image Renderer Component', () => {
  const props = {
    url: 'some_url.jpg',
  };

  let el;
  beforeEach(() => {
    el = shallow(<ImageRenderer {...props} />);
  });
  test('snapshot', () => {
    expect(el).toMatchSnapshot();
  });
});
