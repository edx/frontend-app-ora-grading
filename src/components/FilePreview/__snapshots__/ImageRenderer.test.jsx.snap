// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Renderer Component snapshot 1`] = `
<img
  alt=""
  className="image-renderer"
  src="some_url.jpg"
/>
`;
